### ISSUE TITLE
Migrate to a non clean DB/clean Install magento 2

### ISSUE THREAD

**Issue by loreposh** [magento/data-migration-tool#394]
I want to migrate data but with magento 2 site that theme is already installed.I already tested migrating data to clean install magento 2 but we already have magento 2 that theme is already installed and small changes applied. Is there a way to do this? I get this error: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '1-Migration_Default' for key 'EAV_ATTRIBUTE_  
  SET_ENTITY_TYPE_ID_ATTRIBUTE_SET_NAME'

**Comment by victor-v-rad** [comment]
The tool should not overwrite your theme from Magento 2 during migration. The error `1-Migration_Default` you get because you are trying migration with `-r` option for already migrated store

**Comment by loreposh** [comment]
What I mean is that when you install a theme it has a ready data sql that you'll just import. Already added some ignores and attributes etc. Have tested this migration in a clean magento 2 install and it works but with theme installed and few extension added I still get that "Integrity constraint violation: 1062 Duplicate entry '1-Migration_Default'" and if I put -vvv on migrate:data command it says " Notice: Undefined offset: 1 in /home/magetest/public_html/magento2/vendor/magento/data-migration-tool/src/Migration/Step/Eav/Data.php on line 291 "

Every time I do a data migration command again, I make sure to drop the tables and import the clean database sql with theme data on it.

**Additional error info>-----**

Exception trace:
 () at /home/magetest/public_html/magento2/vendor/magento/framework/App/ErrorHandler.php:61
 Magento\Framework\App\ErrorHandler->handler() at /home/magetest/public_html/magento2/vendor/magento/data-migration-tool/src/Migration/Step/Eav/Data.php:291
 Migration\Step\Eav\Data->migrateAttributeSetsAndGroups() at /home/magetest/public_html/magento2/vendor/magento/data-migration-tool/src/Migration/Step/Eav/Data.php:189
 Migration\Step\Eav\Data->perform() at /home/magetest/public_html/magento2/vendor/magento/data-migration-tool/src/Migration/Mode/AbstractMode.php:73
 Migration\Mode\AbstractMode->runStage() at /home/magetest/public_html/magento2/vendor/magento/data-migration-tool/src/Migration/Mode/Data.php:116
 Migration\Mode\Data->runData() at /home/magetest/public_html/magento2/vendor/magento/data-migration-tool/src/Migration/Mode/Data.php:68
 Migration\Mode\Data->run() at /home/magetest/public_html/magento2/vendor/magento/data-migration-tool/src/Migration/Console/MigrateDataCommand.php:51
 Migration\Console\MigrateDataCommand->execute() at /home/magetest/public_html/magento2/vendor/symfony/console/Symfony/Component/Console/Command/Command.php:257
 Symfony\Component\Console\Command\Command->run() at /home/magetest/public_html/magento2/vendor/symfony/console/Symfony/Component/Console/Application.php:874
 Symfony\Component\Console\Application->doRunCommand() at /home/magetest/public_html/magento2/vendor/symfony/console/Symfony/Component/Console/Application.php:195
 Symfony\Component\Console\Application->doRun() at /home/magetest/public_html/magento2/vendor/magento/framework/Console/Cli.php:96
 Magento\Framework\Console\Cli->doRun() at /home/magetest/public_html/magento2/vendor/symfony/console/Symfony/Component/Console/Application.php:126
 Symfony\Component\Console\Application->run() at /home/magetest/public_html/magento2/bin/magento:23

**Comment by loreposh** [comment]
I re-installed my migration tool, using 2.1.8 version same with my magento 2 version. I already have no source and document mapping errors but I still get error in EAV step in data migration it says "[Exception] Notice: Undefined offset: 152 in /home/mage2test/public_html/magento2/vendor/magento/data-migration-tool/src/Migration/Step/Eav/Data.php on line 809"

I Make sure every time to always drop the tables and import clean sql with theme data before running data migration command again.

Tested it both with or without -r parameter, even if I put -vvv it shows the same error. Last time I did have a data migration success but using a clean install magento. But I want to migrate data to the magento 2 we developed that has theme data already and few extension added. I also added the additional ignore mapping of theme and extensions.

**Comment by loreposh** [comment]
Update on this, I decided to migrate data on a clean magento 2 install and then use 2.2.0 version. I have added some ignores on xml and decided to use the --auto parameter and thus the migration was successful. The only problem is when you go to customers it shows an error now I did experience the same issue in last version that i tested and what i did is add the ignore of the mentioned field in error report to map-customer.xml. But now I can't make it work since the only metadata field i can see is 'firstname'.

Sample error:

{"0":"Unable to unserialize value.","1":"#0 \/home\/mageserver\/public_html\/magento2\/app\/code\/Magento\/Eav\/Model\/Attribute.php(109): Magento\\Framework\\Serialize\\Serializer\\Json->unserialize('a:2:{s:15:\"max_...')\n#1 \/home\/mageserver\/public_html\/magento2\/app\/code\/Magento\/Customer\/Model\/AttributeMetadataConverter.php(90): Magento\\Eav\\Model\\Attribute->getValidateRules()\n#2 \/home\/mageserver\/public_html\/magento2\/app\/code\/Magento\/Customer\/Model\/Metadata\/CustomerMetadata.php(77): Magento\\Customer\\Model\\AttributeMetadataConverter->createMetadataAttribute(Object(Magento\\Customer\\Model\\Attribute))\n#3 \/home\/mageserver\/public_html\/magento2\/app\/code\/Magento\/Customer\/Model\/Metadata\/CustomerMetadata.php(109): Magento\\Customer\\Model\\Metadata\\CustomerMetadata->getAttributeMetadata('firstname')\n#4 \/home\/mageserver\/public_html\/magento2\/app\/code\/Magento\/Customer\/Model\/Metadata\/CachedMetadata.php(85): Magento\\Customer\\Model\\Metadata\\CustomerMetadata

**Comment by victor-v-rad** [comment]
Hi @loreposh 

It seems records were not converted into JOSN in this filed customer_eav_attribute.validate_rules 
See map-eav.xml.dist 
```
            <transform>
                <field>customer_eav_attribute.validate_rules</field>
                <handler class="\Migration\Handler\SerializeToJson"/>
            </transform>
```

**Comment by PascalBrouwers** [comment]
I had this for weee_tax_applied for 1.9.2.4 and it seemed missing in the map.xml.dst

**Comment by ghost** [comment]
Just a warning:
After migration (1.9.2.4 to 2.3.0) I changed the attribteset ánd content by removing this prefix. So I also changed "Migration_General" to Genral. Works fine, until.... some extension wants to implement an attribute in the General map. Then upgrading database gave trouble. 
Also, genrating the "General" tab/map within an attrubuteset, moving attributes to it and deleting the "Migration_General" will cause errors.
In short, I'm not gonna border about this Migration_ prefix. When it works, it works :)
